common package time update

AvaAvarai opened the pull request:
- Swap DateRange for Interval from date-fns
- getWeekRangeOfDate function update
- Shift month indexing to match date-fns 0 start
- Programmatic testing of getWeekRangeOfDate (needs more cases)
- Constants for start and end of week

tuckersn commented:
Hey Ava,

Good work! Just a few things before we merge though.
- Some files are using spaces instead of tabs.
- Why'd you remove the enum for days of the week? I wrote the enum mainly as a string to number conversion for date-fns's days of the week number. I also like the enum because we can use it in the setDay function as-is.

AvaAvarai commented:
Thank you for the feedback, very much appreciated! With the new commits:

- I've switched to tabs instead of spaces
- Added the enum back and used to set start/end of week placeholder
- Used a helper func from date-fns: lastDayOfWeek for gettingWeekRangeOfDate, allowing for dynamic week starts here
